### PR TITLE
openeo GRASS driver: support actinia processes

### PR DESCRIPTION
**Pull request by metzm**
add generic translation of actinia processes from openeo pg to actinia pg

waiting for https://github.com/mundialis/actinia-gdi/pull/44

### CODE REVIEW

**Comment by neteler** [comment]
(https://github.com/mundialis/actinia-gdi/pull/44 has been merged)

File: src/openeo_grass_gis_driver/actinia_processing/base.py [hunk 1]
```diff
@@ -271,6 +276,141 @@ def process_node_to_actinia_process_chain(node: Node) -> Tuple[list, list]:
...
+                datatype = GrassDataType.STRDS
+
+            if datatype is not None:
+                # note that key is added to the output name
+                # in order to distinguish between different outputs
+                # of the same module
+                output_object = DataObject(name=f"{data_object.name}_{process_name}_{key}", datatype=datatype)
+                param = {"param": key,
+                         "value": output_object.grass_name()}
+                pc["inputs"].append(param)
```
Comment (mmacata) [inline src/openeo_grass_gis_driver/actinia_processing/base.py:405]: you could use pc["outputs"] for that.

**Review by mmacata — COMMENTED** [review]
Great, thank you! A good step forward in exploiting all GRASS GIS functionalty, if not THE step.
I only have some minor remarks.

File: tests/test_jobs.py [hunk 2]
```diff
@@ -94,6 +94,28 @@ def test_job_creation_deletion_1(self):
         response = self.app.get(f'/jobs/{job_id}', headers=self.auth)
         self.assertEqual(404, response.status_code)
 
+    def test_actinia_process(self):
+        """Run the test in the ephemeral database
+        """
+        JOB_TEMPLATE["process"] = ACTINIA_PROCESS["process"]
```
[the remark this answers is not in the document]
Comment (metzm) [reply tests/test_jobs.py:100]: Let's change it later.

File: src/openeo_grass_gis_driver/actinia_processing/base.py [hunk 3]
```diff
@@ -271,6 +276,141 @@ def process_node_to_actinia_process_chain(node: Node) -> Tuple[list, list]:
...
+            # warning, error, exception?
+            continue
+        # check if it is an input object
+        if isinstance(node.arguments[key], dict) and \
+           "from_node" in node.arguments[key]:
+            # input option comes from another node in the process graph
+	    # which output object in the set of output_objects?
+            value = list(node.get_parent_by_name(parent_name=key).output_objects)[0]
+            data_object = value
+        elif item["schema"]["type"] == "boolean":
```
[the remark this answers is not in the document]
Comment (metzm) [reply src/openeo_grass_gis_driver/actinia_processing/base.py:353]: Thanks, done.

File: src/openeo_grass_gis_driver/actinia_processing/base.py (hunk 1 above)
Comment (metzm) [reply src/openeo_grass_gis_driver/actinia_processing/base.py:405]: `pc["outputs"]` is not used in any of the openeo process translations. Worth a separate ticket?

Comment (mmacata) [reply src/openeo_grass_gis_driver/actinia_processing/base.py:405]: Wouldn't change function, only semantic. I resolve this thread.

**Review by mmacata — APPROVED** [review]

**Pull request merged**